### PR TITLE
Requirement Documentation

### PR DESCRIPTION
**Pull request by c0nD**
Solves issue #20 for requirement documentation.

- Created docs folder
- Added REQFILE.md to supply documentation

### CODE REVIEW

File: docs/REQFILE.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+5. Each cookie can accept a collection of **validity requirements** and one **objective requirement**.
+
+## Bot Moderator Permissions and Capabilities
+
+### Actions on Behalf of Others
+1. **Uploading Requirements for Users**: Moderators can upload requirement files directly into other users' inventories.
+2. **Optimizing Using User's Topping Inventory**: Moderators have the capability to optimize configurations using the topping inventory of other users.
+
+### Copying Requirement Files
+1. **Copying Files**: Moderators can copy requirement files between users, ensuring efficient management and sharing of configurations within the guild.
```
Comment (wumphlett) [inline docs/REQFILE.md:19]: copying can be done by both mods & non-mods, but it looks different:
Non-mod, copy a default req into a private copy
Mod: copy one of your private reqs into someone elses private reqs

File: docs/REQFILE.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+- **Range**: Specify a range of valid substat values.
+  - Example: `28 <= Cooldown <= 28.5`
+- **Equality**: Pin a substat to a specific value.
+  - Example: `Cooldown == 28.3`
+
+## Objective
+Objective requirements define a goal to achieve with the cookie's stats:
+- **Objective**: Specify a goal with a substat.
+- **Special**: Include special requirements with relevant information.
+  - `E[DMG]`: Combines CRIT% and ATK.
```
Comment (wumphlett) [inline docs/REQFILE.md:40]: Also the `Vitality` objective, combines DMG Resistance and HP into the best possible chance for survival

Comment (wumphlett) [inline docs/REQFILE.md:40]: Might also be good to specify that its not a simple combo, it does the math for which is the right mix of the two to maximize expected dmg

File: docs/REQFILE.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+- **Types**: Various stat modifiers can be included such as CRIT%, CRIT DMG, ATK MULT, etc.
+- **Specification**: Each modifier should specify its source and value.
+
+## Discord Commands
+- **Upload a file**: `!req upload`
+  - Attach your .yaml file to the Discord message.
+  - Use `!req upload [user]` to upload a file for someone else. If no user is specified, it will automatically add the reqfile to you.
+- **View files**: `!req view`
+  - Displays all of your uploaded requirement files.
+- **Copy a file**: `!req copy [user]`
```
Comment (wumphlett) [inline docs/REQFILE.md:55]: Same as above, only mods need to specify a user. All of the req commands that allow an optional target assume that the target is specified by a mod

File: docs/REQFILE.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+- name: Werewolf
+  requirements:
+  - Cooldown >= 9
+  - DMG Resist >= 10
+  - ATK SPD >= 7.5
+  - max: CRIT%
+- name: Rye
+  requirements:
+  - max: E[DMG]
+    ATK: 33.72
```
Comment (wumphlett) [inline docs/REQFILE.md:81]: I think we should explain that this is a cookie's base atk% buff, calculable with `!basestat`

**Comment by c0nD** [comment]
Should have resolved all comments.

Commit history looks a bit weird.. had a `git add .` in the middle so I had to change the HEAD pointer back a few commits to fix other file changes.

Let me know how the changes look.

**Review by wumphlett — APPROVED** [review]
LGTM

**Pull request merged**